examples/bigquery-analyze-realtime-reddit-data/terraform/main.tf

```diff
@@ -0,0 +1,284 @@
+  role    = "roles/pubsub.admin"
+  members = [
+    "serviceAccount:${google_service_account.reddit_service_account.email}"
+  ]
+}
+
+
+resource "google_compute_instance" "reddit_vm" {
+  name         = "reddit-data-collector-1"
+  machine_type = "e2-micro"
+  zone         = "us-central1-a"
+
+
+  boot_disk {
+    initialize_params {
+      image = "centos-cloud/centos-7"
+    }
+  }
+
+  network_interface {
+    network = "default"
+
+    access_config {
+      // Ephemeral public IP
+    }
+  }
+
+  metadata_startup_script = <<EOF
+    echo \"install python and git...\"
+    yum install python3 -y
+    yum install git -y
+    echo \"installing pip...\"
+    curl https://bootstrap.pypa.io/pip/3.6/get-pip.py -o get-pip.py
+    python3 get-pip.py
+    python3 -m pip install --upgrade pip
+    echo \"importing python libraries...\"
+    python3 -m pip install praw
+    python3 -m pip install pandas
+    python3 -m pip install textblob
+    python3 -m pip install better_profanity
+    python3 -m pip install google-cloud-dataflow-client
+    python3 -m pip install textstat
+    python3 -m textblob.download_corpora
+    python3 -m pip install google-cloud-pubsub
+    echo \"downloads complete...\"
+    echo \"creating dirs for git repo...\"
+    mkdir reddit
+    cd reddit
+    git init
+    git clone https://github.com/CYarros10/reddit-streaming-application.git
+    echo \"git clone complete...\"
+    cd reddit-streaming-application/app/
+    echo \"current dir:\"
+    pwd
+    echo \"changing permissions\"
+    chmod 777 comment-stream-detailed-gcp.py
+    chmod 777 praw.ini
+    echo \"updating scripts with custom info\"
+    sed -i -r 's/<insert-project-id-here>/${var.project_id}/g' comment-stream-detailed-gcp.py
+    sed -i -r 's/<insert-topic-id-here>/${var.pubsub_topic_name}/g' comment-stream-detailed-gcp.py
+    sed -i 's/<insert-client-id-here>/${var.reddit_client_id}/g' praw.ini
+    sed -i 's/<insert-client-secret-here>/${var.reddit_client_secret}/g' praw.ini
+    sed -i 's/<insert-username-here>/${var.reddit_username}/g' praw.ini
+    sed -i 's/<insert-password-here>/${var.reddit_password}/g' praw.ini
+    echo \"starting stream\"
+    python3 comment-stream-detailed-gcp.py funny askreddit todayilearned science worldnews pics iama gaming videos movies aww blog music news explainlikeimfive askscience books television mildlyinteresting lifeprotips space showerthoughts diy jokes sports gadgets nottheonion internetisbeautiful photoshopbattles food history futurology documentaries dataisbeautiful listentothis upliftingnews personalfinance getmotivated oldschool cool philosophy art writingprompts fitness technology bestof adviceanimals politics atheism europe &>> logs.txt
+  EOF
+
+  service_account {
+    # Google recommends custom service accounts that have cloud-platform scope and permissions granted via IAM Roles.
+    email  = "${google_service_account.reddit_service_account.email}"
+    scopes = ["cloud-platform"]
+  }
+}
```

Comment:
you might want to add a depends_on on the google_project_iam_binding resources this VM is relying on. It probably doesn't cause an error since, the VM creation probably takes longer then the parallel iam binding creation. But it's more explicit.